### PR TITLE
refactor: anyhow usage

### PR DESCRIPTION
**Pull request by aatifsyed**
# Changes
- `opt.ok_or_else(|| anyhow::anyhow!("foo"))` -> `opt.context("foo")`
- `opt.ok_or_else(|| anyhow::anyhow!("foo {}", bar))` -> `opt.with_context(||format!("foo {}", bar))`
- `#[test] fn foo() -> anyhow::Result<()>` -> `#[test] fn foo()`
- Rationalise `anyhow` imports

### CODE REVIEW

File: tests/keystore_tests.rs [hunk 1]
```diff
@@ -13,8 +12,8 @@ use crate::common::{create_tmp_config, daemon, CommonArgs};
 
 // https://github.com/ChainSafe/forest/issues/2499
 #[test]
-fn forest_headless_encrypt_keystore_no_passphrase_should_fail() -> Result<()> {
-    let (config_file, _data_dir) = create_tmp_config()?;
+fn forest_headless_encrypt_keystore_no_passphrase_should_fail() -> anyhow::Result<()> {
```
Comment (lemmih) [inline tests/keystore_tests.rs:15]: Did you overlook these tests or do you really want them to return a `Result`?

```suggestion
fn forest_headless_encrypt_keystore_no_passphrase_should_fail() {
```

Comment (aatifsyed) [reply tests/keystore_tests.rs:15]: Finished an updated the description

**Review by lemmih — APPROVED** [review]

**Review by ruseinov — APPROVED** [review]
looks great!

**Comment by ruseinov** [comment]
Wait, still got some `anyhow::Result` in tests.

File: tests/keystore_tests.rs (hunk 1 above)
Comment (ruseinov) [reply tests/keystore_tests.rs:15]: I'm not sure what happened here

**Comment by ruseinov** [comment]
Seems good to me, just want to understand why there is a bunch of tests that retains `anyhow::Result` as return type:
1. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/import_snapshot_tests.rs
2. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/keystore_tests.rs
3. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/config.rs

There is also a bunch of places where we have `Result<(), anyhow::Error>` as opposed to `anyhow::Result<()>`. I'm not sure if this should be cleaned up or not, but simply searching for `-> Result` gives enough evidence of the latter.

**Comment by aatifsyed** [comment]
> Seems good to me, just want to understand why there is a bunch of tests that retains `anyhow::Result` as return type:
> 
> 1. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/import_snapshot_tests.rs
> 2. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/keystore_tests.rs
> 3. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/config.rs
> 
> There is also a bunch of places where we have `Result<(), anyhow::Error>` as opposed to `anyhow::Result<()>`. I'm not sure if this should be cleaned up or not, but simply searching for `-> Result` gives enough evidence of the latter.

Ah, I only searched in `src/` :)

**Comment by ruseinov** [comment]
> > Seems good to me, just want to understand why there is a bunch of tests that retains `anyhow::Result` as return type:
> > 
> > 1. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/import_snapshot_tests.rs
> > 2. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/keystore_tests.rs
> > 3. https://github.com/ChainSafe/forest/blob/2c69e5c0f4d22f69e7dad1771f5c129bb284bd48/tests/config.rs
> > 
> > There is also a bunch of places where we have `Result<(), anyhow::Error>` as opposed to `anyhow::Result<()>`. I'm not sure if this should be cleaned up or not, but simply searching for `-> Result` gives enough evidence of the latter.
> 
> Ah, I only searched in `src/` :)

Made a mistake in my comment about `Result<(), anyhow::Error>`. `latter -> former`. As in I have found some occurrences.
`

**Comment by ruseinov** [comment]
I think this is still missing the refactoring of cases where we have `-> Result<..., anyhow::Error>`. We should be using `anyhow::Result<...>` instead. 
Try searching for `-> Result` to find occurrences.

**Comment by LesnyRumcajs** [comment]
How do we prevent this in the future? Reviewers should focus on the functional aspects of PRs, not things that should be basically lints.

**Comment by aatifsyed** [comment]
> How do we prevent this in the future? Reviewers should focus on the functional aspects of PRs, not things that should be basically lints.

I've written a semgrep tool that catches this - that's why I missed the occurences in `tests/` - I was just looking in `src/`

We can check that in later

**Comment by LesnyRumcajs** [comment]
> > How do we prevent this in the future? Reviewers should focus on the functional aspects of PRs, not things that should be basically lints.
> 
> I've written a semgrep tool that catches this - that's why I missed the occurences in `tests/` - I was just looking in `src/`
> 
> We can check that in later

If we don't forget it and actually plan to do it, then sure.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**